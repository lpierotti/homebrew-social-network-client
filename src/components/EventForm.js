import React from 'react'
import { Form, Label } from 'semantic-ui-react'
import DatePicker from 'react-datepicker';
import moment from 'moment';
import { connect } from 'react-redux'
import { createEvent } from '../actions/events'
import Select from 'react-select'

import 'react-datepicker/dist/react-datepicker.css';
import 'react-datepicker/dist/react-datepicker-cssmodules.css'


class EventForm extends React.Component {

	constructor() {
		super()
		this.state = {
			name: '',
			description: '',
			number: '',
			street: '',
			city: '',
			state: '',
			start: moment(),
			end: moment(), 
			submitted: false
		}
	}

	handleChange = (event) => {
		this.setState({[event.target.name]: event.target.value})
	}

	handleStartDateChange = (date) => {
		this.setState({start: date})
	}

	handleEndDateChange = (date) => {
		this.setState({end: date})
	}

	handleSelect = (selected) => {
		this.setState({state: selected.value})
	}

	handleSubmit = (event) => {
		event.preventDefault()
		this.props.createEvent({...this.state, id: this.props.id})
		this.setState({submitted: true})
	}

	render() {
		var options = ["AK","AL","AR","AZ","CA","CO","CT","DC","DE","FL","GA","GU","HI","IA","ID", "IL","IN","KS","KY","LA","MA","MD","ME","MH","MI","MN","MO","MS","MT","NC","ND","NE","NH","NJ","NM","NV","NY", "OH","OK","OR","PA","PR","PW","RI","SC","SD","TN","TX","UT","VA","VI","VT","WA","WI","WV","WY"].map(abbreviation => ({value: abbreviation, label: abbreviation}))
		if (this.state.submitted) {
			return <div style={{minHeight: '500px'}}><h3 style={{margin:'auto'}}>{this.state.name} was created</h3></div>
		}
		return (
			<div style={{minHeight: '500px'}}>
				<h2 style={{textAlign: 'center'}}>Create an Event</h2>
				<Form style={{maxWidth: '400px', margin: 'auto'}}onSubmit={this.handleSubmit}>
					<Form.Group>
						<Form.Input placeholder='Name'name='name' type='text' onChange={this.handleChange}/>
						<Form.Input placeholder='Description'name='description' type='text' onChange={this.handleChange}/>
					</Form.Group>
					<Form.Group>
						<Form.Input placeholder='Address Number'name='number' type='text' onChange={this.handleChange}/>
						<Form.Input placeholder='Street'name='street' type='text' onChange={this.handleChange}/>
					</Form.Group>
					<Form.Group>
						<Form.Input placeholder='City'name='city' type='text' onChange={this.handleChange}/>
						<Select 
							options={options}
							onChange={this.handleSelect}
							placeholder={'State'}
						/>
					</Form.Group>
					<Label>Start Time</Label>
					<DatePicker name='start' 
						onChange={this.handleStartDateChange}
						selected={this.state.start}
						showTimeSelect
					    timeIntervals={15}
					    dateFormat="LLL"
					    minDate={moment()}
					/>
					<Label>End Time</Label>
					<DatePicker name='end' 
						onChange={this.handleEndDateChange}
						selected={this.state.end}
						showTimeSelect
					    timeIntervals={15}
					    dateFormat="LLL"
					    minDate={moment()}
					/>
					<Form.Button>Submit</Form.Button>
				</Form>
			</div>

		)
	}
}

function mapDispatchToProps(dispatch) {
	return {
		createEvent: (params) => dispatch(createEvent(params))
	}
}

export default connect(null, mapDispatchToProps)(EventForm)

