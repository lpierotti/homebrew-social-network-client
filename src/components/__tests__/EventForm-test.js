import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { EventForm } from '../EventForm';
import renderer from 'react-test-renderer';
import Provider from 'react-redux'

describe('EventForm component', () => {

	it('renders correctly', () => {
		const tree = renderer.create(
			<EventForm />
		).toJSON();
		expect(tree).toMatchSnapshot();
	})
})