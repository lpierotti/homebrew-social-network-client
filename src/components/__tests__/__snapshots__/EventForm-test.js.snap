// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm component renders correctly 1`] = `
<div
  style={
    Object {
      "minHeight": "500px",
    }
  }
>
  <h2
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Create an Event
  </h2>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
    style={
      Object {
        "margin": "auto",
        "maxWidth": "400px",
      }
    }
  >
    <div
      className="fields"
    >
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            disabled={undefined}
            name="name"
            onChange={[Function]}
            placeholder="Name"
            required={undefined}
            tabIndex={undefined}
            type="text"
          />
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            disabled={undefined}
            name="description"
            onChange={[Function]}
            placeholder="Description"
            required={undefined}
            tabIndex={undefined}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            disabled={undefined}
            name="number"
            onChange={[Function]}
            placeholder="Address Number"
            required={undefined}
            tabIndex={undefined}
            type="text"
          />
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            disabled={undefined}
            name="street"
            onChange={[Function]}
            placeholder="Street"
            required={undefined}
            tabIndex={undefined}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="fields"
    >
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            disabled={undefined}
            name="city"
            onChange={[Function]}
            placeholder="City"
            required={undefined}
            tabIndex={undefined}
            type="text"
          />
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            disabled={undefined}
            name="state"
            onChange={[Function]}
            placeholder="State"
            required={undefined}
            tabIndex={undefined}
            type="text"
          />
        </div>
      </div>
    </div>
    <div
      className="ui label"
      onClick={[Function]}
    >
      Start Time
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className=""
            disabled={false}
            id={undefined}
            name="start"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value="November 20, 2017 5:05 PM"
          />
        </div>
      </div>
    </div>
    <div
      className="ui label"
      onClick={[Function]}
    >
      End Time
    </div>
    <div>
      <div
        className="react-datepicker-wrapper"
      >
        <div
          className="react-datepicker__input-container"
        >
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            className=""
            disabled={false}
            id={undefined}
            name="end"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            tabIndex={undefined}
            title={undefined}
            type="text"
            value="November 20, 2017 5:05 PM"
          />
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <button
        className="ui button"
        disabled={undefined}
        onClick={[Function]}
        required={undefined}
        role="button"
        tabIndex={undefined}
        type={undefined}
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
